runelite-client/src/main/java/net/runelite/client/plugins/twitch/TwitchPlugin.java

```diff
@@ -0,0 +1,421 @@
+	{
+		twitchShutdown(irc);
+		irc = twitchSetup(config.oauthToken(), config.channelName(), config.nickName());
+	}
+
+	@Override
+	protected void shutDown()
+	{
+		twitchShutdown(irc);
+	}
+
+	@Subscribe
+	public void onConfigChanged(ConfigChanged event)
+	{
+		if (!event.getGroup().equals("twitch"))
+		{
+			return;
+		}
+
+		if (event.getKey().equals("channel") || event.getKey().equals("oauth") || event.getKey().equals("nick"))
+		{
+			twitchShutdown(irc);
+			irc = twitchSetup(config.oauthToken(), config.channelName(), config.nickName());
+		}
+	}
+
+	private boolean isChatboxTransparent()
+	{
+		return client.isResized() && client.getVar(Varbits.TRANSPARENT_CHATBOX) == 1;
+	}
+
+	private Client twitchSetup(String oauth, String nick, String name)
+	{
+		Client ircClient = Client.builder()
+				.serverHost("irc.chat.twitch.tv").serverPort(443)
+				.serverPassword(oauth)
+				.nick(nick)
+				.messageSendingQueueSupplier(TwitchDelaySender.getSupplier(false))
+				.build();
+		TwitchListener listener = new TwitchListener(ircClient);
+		ircClient.getEventManager().registerEventListener(listener);
+		ircClient.getEventManager().registerEventListener(this);
+		ircClient.connect();
+		ircClient.addChannel("#" + name);
+		return ircClient;
+	}
+
+	//Send messages to twitch chat using command "t"
+	@Subscribe
+	private void onCommand(CommandExecuted executed)
+	{
+		if (!client.getGameState().equals(GameState.LOGGED_IN))
+		{
+			return;
+		}
+
+		String[] args = executed.getArguments();
+		StringBuilder sb = new StringBuilder();
+		String command = executed.getCommand();
+
+		if (command.equalsIgnoreCase("t"))
+		{
+			for (String arg : args)
+			{
+				sb.append(arg + " ");
+			}
+
+			String message = sb.toString();
+			irc.sendMessage("#" + config.nickName(), message);
+			//Echos the message back in in-game chat
+			if (irc.getChannels().toString().contains("#" + config.nickName()))
+			{
+				//If the chat is in sub only mode and they are not a subscriber
+				if (!subscribed && subsOnly)
+				{
+					client.addChatMessage(ChatMessageType.TRADE, "",
+							"<col=ffffff></col>[<col=9070ff>Twitch</col>] Notification: <col=3D2BAD>This channel is in Subscriber only mode!</col>", "");
+				}
+				//If chat is in emote only mode
+				else if (emoteOnly && !broadcaster && !mod)
+				{
+					client.addChatMessage(ChatMessageType.TRADE, "",
+							"<col=000000></col>[<col=0000FF>Twitch</col>] Notification: <col=3D2BAD>This channel is in emote only mode!<br><col=3D2BAD>Function coming soon to a Runelite client near you.</col>", "Twitch");
+				}
+				else
+				{
+					sendMessage(message, badges + config.channelName());
+				}
+			}
+		}
+	}
+
+	private void sendMessage(String message, String name)
+	{
+		String oColor = Integer.toHexString(config.chatOColor().getRGB() & 0xFFFFFF);
+		String tColor = Integer.toHexString(config.chatTColor().getRGB() & 0xFFFFFF);
+
+		if (isChatboxTransparent())
+		{
+			client.addChatMessage(ChatMessageType.TRADE, "",
```

Comment:
Again I'd probably store these in a constant, though it may be worth going a step further and having a method that you pass colour, name and message to, and it returns the built string.